source/common/upstream/ring_hash_lb.cc

```diff
@@ -118,12 +130,36 @@ void RingHashLoadBalancer::Ring::create(
       config.valid()
           ? PROTOBUF_GET_WRAPPED_OR_DEFAULT(config.value().deprecated_v1(), use_std_hash, true)
           : true;
+
+  std::unique_ptr<char[]> hash_key_buffer;
+  uint64_t last_hash_key_size = 0;
   for (const auto& host : hosts) {
+    const std::string& address_string = host->address()->asString();
+    uint64_t offset_start = address_string.size();
+
+    // Although in almost all cases the buffer could be stack allocated to 128 bytes or so, we
+    // don't explicitly know what is in the address string (e.g., UDS path). We allocate a raw
+    // buffer on the heap and use it for all calculations. The needed size is the sizse of the
+    // address, plus '_', plus 32 bytes for the index. All of this is done to avoid string
```

Comment:
I think if we separate out this optimization work it could help derisk the PR. Maybe do an earlier PR which pulls out this and some of the other independent optimizations/changes (e.g. string_view)?

Reply:
Yup I can remove this optimization and do it in a follow up.